### ISSUE TITLE
Refactor Odyssey to support WA state sites

### ISSUE THREAD

**Issue by zstumgoren** [biglocalnews/court-scraper#60]
> in lieu of registered access to WA Odyssey sites, we should implement #72 and test against open CAPTCHA-based sites

A number of WA state counties use Odyssey.  The WA sites (we discovered on Snohomish) revealed a bug due to difference case metadata on search result pages, compared to metadata present on GA Dekalb/Chatham.

Need to refactor code to be more flexible for Odyssey search result pages.

## Reproducing the bug

To reproduce the bug:

```python
from court_scraper.platforms.odyssey_site import OdysseySite

url = "https://prdodypub.courts.wa.gov/ODYPORTAL/Home/Dashboard/29"
site = OdysseySite(url, "user@email.com", "PASSWORD", download_dir='/tmp')
site.login(headless=False)
results = site.search(search_terms=["19-2-11450-31"], headless=False)
```

## Problem locale

```python
# court_scraper/platforms/odyssey_site/pages/search_results.py
     64     def metadata(self):
     65         case_detail_url = self.el.find_element(
     66                 *SearchResultsPageLocators.CASE_DETAIL_LINK
     67             ).get_attribute('data-url')
     68         # FIX NEEDED HERE
     69         # Probably more flexible handling based on known column headers
     70         case_num, style_def, file_date, status, party_name = [
---> 71             field.strip() for field in self.inner_text.split('\t')
     72         ]
     73         return {

```

## Stacktrace

```
Traceback (most recent call last):
  File "wa_portal.py", line 6, in <module>
    results = site.search(search_terms=["19-2-11450-31"], headless=False)
  File "/Users/tumgoren/code/stanford/biglocal/court-scraper/court_scraper/platforms/odyssey_site/site.py", line 44, in search
    row_data = case_row.metadata
  File "/Users/tumgoren/code/stanford/biglocal/court-scraper/court_scraper/platforms/odyssey_site/pages/search_results.py", line 69, in metadata
    field.strip() for field in self.inner_text.split('\t')
ValueError: not enough values to unpack (expected 5, got 4)

```

**Comment by zstumgoren** [comment]
The fix for #69 required more flexible handling of header rows, which should in theory set us up to handle the variability in Washington. We'll need to ensure that the CaseInfo mappings (the raw field names to standardized field names) are handled for whatever is available in WA

**Comment by zstumgoren** [comment]
@justinmayo Note that both of our accounts appear to have been locked by this portal, so I'm unable to test with your credentials or mine. Not sure if this is a recent development, but it appears the portal is CAPTCHA protected for unregistered users and requires you to [contact court administrators to register](https://www.courts.wa.gov/index.cfm?fa=home.sub&org=sccms&page=OdysseyPortalRegister) for access to documents (and potentially other, more sensitive data?).
